Devel environment

zergin opened the pull request:
As I'd like to push doThings forward a bit to scratch some of my itches here is a PR for development environment. 

This adds docker-compose support creating three containers: nginx, php and a gulp builder. Each container adheres to "single-process" mantra. The PHP container is run as unprivileged user. Also composer dependencies are updated so that there are no fatal errors (and deprecation warnings) during install. 

**NOTE:** Entire environment is wired for development (app_dev.php Symfony front controller), host mounted volumes with sources, gulp builder watching. 

To run:

```
docker-compose up
```

To stop and clean:

```
docker-compose down
```

Docker-compose 1.7+ is required (and docker-engine 1.11+) because of v2 configuration file.

zergin committed:
Update dependencies

- switch to ^3.0 for sensio dependencies forced by current composer
  (match Event classes)

zergin committed:
Dockerize development environment

- This adds definitions for docker (1.11+) based development environment.
- Orchestration via docker-comppse (1.7+).
- Taskwarrior built from source: 2.5.1.

* To run: `docker-compose up` and go http://localhost:8000/
* To stop: `docker-compose down`

See more at: https://docs.docker.com/compose/overview/

NOTE: All configuration is wired for development. Not suited for production
deployment.

DavidBadura commented on composer.json:
```diff
@@ -14,8 +14,8 @@
-        "sensio/distribution-bundle": "^3.0.12",
-        "sensio/framework-extra-bundle": "^3.0.2",
+        "sensio/distribution-bundle": "^3.0",
+        "sensio/framework-extra-bundle": "^3.0",
```
why did you change that?

zergin replied:
Because of current composer that requires distribution-bundle afaik > 3.0.18 (event classes changes). I had post-install script errors w/o this change.

Do you see any problem with locking to 3.0.x? It seems to be working fine for me in preliminary tests.

DavidBadura commented on bin/runner.sh:
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash -x
+
+npm install -g gulp-cli
```
you don't need gulp-cli

zergin replied:
removed.

DavidBadura commented:
otherwise :+1:

zergin commented:
I've removed adding gulp-cli (a leftover from running gulp directly before I saw "scripts" definition in package.json).

I also changed Dockerfile a bit not to include vendors while building. As this is purely development image all requirements must be installed after host volume is mounted anyway (call to `composer install` is done on every container start).

DavidBadura commented:
thank you :smiley:
